torch/csrc/Exceptions.h

```diff
@@ -14,6 +13,10 @@
 #define END_HANDLE_TH_ERRORS_RET(retval)                                       \
   } catch (python_error &e) {                                                  \
     return retval;                                                             \
+  } catch (torch::PyTorchError &e) {                                           \
```

Comment:
a nicer solution would be to call it `set_python_error` or rename `python_error` to `error_already_set` (like `pybind` does)

Reply:
Either is fine with me. Which do you prefer?

Reply:
I'm also fine with both of them. `error_already_set` seems a bit cleaner, but it's a larger change so I understand if you don't want to do it